src/Quiz.js

```diff
@@ -1,208 +1,168 @@
-		} else {
-			delete incorrectCounter[quizId][question];
-		}
+const INCORRECT_COUNTER_LOCAL_STORAGE_KEY = 'incorrectCounter';
 
-		console.log(incorrectCounter[quizId]);
+/*
+
+THINGS I'M UNSURE ABOUT (from the class->function+hooks refactor)
+1. a)
+	 Where to put functions now that they're no longer methods? (At the moment, I have half inside and half outside because wasn't sure)
+   Outside the function-based component? Inside the component? 
+	 If they need to use setState function inside another function, should I just pass these in as arguments?
+
+
+	 1 b)
+	 Is it bad practice for nested functions to rely on state variables (in the outer scope?), or should they be turned into parameters?
+	 Eg. handleAnswerSubmit uses currentQuestionIdx (which is a state variable)
+
+2. Updating state based on previous state. Is it guaranteed that the previous state will have already updated?
+	 Eg. in updateScore function, replacing
+	        this.setState((curSt) => ({ correct: curSt.correct + 1 })); 
+		with 
+		      setCorrect(currentCorrect + 1)
+		     
+3. Am I using useEffect in the correct way? (ie. as a replacement for componentDidMount)
+
+4. Do I have too many state variables?
+
+*/
+
+function Quiz(props) {
+	const { quizProps } = props;
+	const { quizId } = quizProps;
+	const [ questionsAll, setQuestionsAll ] = useState([]);
+	const [ questions, setQuestions ] = useState([]); // might be better (ie. more space efficient) to just store the desired indices
+	const [ loadingData, setLoadingData ] = useState(true);
+	const [ correct, setCorrect ] = useState(0);
+	const [ incorrect, setIncorrect ] = useState(0);
+	const [ currentQuestionIdx, setCurrentQuestionIdx ] = useState(0);
+	const [ repeatCorrectAnswerMode, setRepeatCorrectAnswerMode ] = useState(false);
+	const [ practiceMode, setPracticeMode ] = useState(true);
+	const [ answerStatus, setAnswerStatus ] = useState('none');
+	const [ onlyPractiseIncorrect, setOnlyPractiseIncorrect ] = useState(false);
+
+	useEffect(() => {
+		console.log('useEffect');
+
+		const getQuestions = async () => {
+			const allQuestions = await quizProps.questionGetter();
+			console.log('allQuestions', allQuestions);
+			let questionsFiltered = getFilteredQuestions(allQuestions);
+			if (onlyPractiseIncorrect) {
+				const incorrectCounts = getQuestionsWithIncorrectCounts(
+					INCORRECT_COUNTER_LOCAL_STORAGE_KEY
+				);
+				questionsFiltered = allQuestions.filter((q) =>
+					Object.keys(incorrectCounts[quizId]).includes(q.question)
+				);
+			}
 
-		window.localStorage.setItem(
-			INCORRECT_COUNTER_LOCAL_STORAGE_KEY,
-			JSON.stringify(incorrectCounter)
-		);
-	}
+			// should set state inside the async function according to : https://devtrium.com/posts/async-functions-useeffect
+			setQuestionsAll(allQuestions);
+			setQuestions(questionsFiltered);
+			setLoadingData(false);
+		};
+		getQuestions(); // todo need to catch this?
+	}, []); // when should useEffect be called??
```

Comment:
I would rephrase the thinking to when should I *let* this run. useEffect is happy to run on every render if you let it. Your dependency array (the second parameter to useEffect aka your empty array) is the gatekeeper that prevents that from happening. 

So the answer to when should I *let* this run, is when you have something (usually a state variable) that was updated and you need to rerun this code with the new state variable. Let's say you needed to refetch the questions if a user swaps between practice mode and quiz mode. You'd pass one of the state variables there and determine what to do when it changes.